saleor/checkout/checkout_cleaner.py

```diff
@@ -70,12 +75,19 @@ def clean_checkout_payment(
     lines: Iterable["CheckoutLineInfo"],
     discounts: Iterable[DiscountInfo],
     error_code: Type[CheckoutErrorCode],
-    last_payment: Optional[payment_models.Payment],
+    current_payment: Optional[payment_models.Payment],
 ):
     clean_billing_address(checkout_info, error_code)
-    if not is_fully_paid(manager, checkout_info, lines, discounts):
-        gateway.payment_refund_or_void(
-            last_payment, manager, channel_slug=checkout_info.channel.slug
+
+    if len(get_active_payments(checkout_info.checkout)) == 0:
```

Comment:
Would be better to call `.count()` on db level instead of retrieving all objects and calling `len`

Reply:
I added prefetch at the top level: https://github.com/mirumee/saleor/pull/8001/commits/b53df6cccf0786f92164d8db540ac6d8418fb71d.

`get_active_payments` is being called several times during processing of this request and it's probably cheaper to refer to that qs rather than hitting the database once again.